packages/zoe/src/zoeService/zoe.js

```diff
@@ -427,52 +468,30 @@ function makeZoe(vatAdminSvc, zcfBundleName = undefined) {
             }
           });
 
-          return Promise.all(paymentDepositedPs).then(amountsArray => {
+          return Promise.all(paymentDepositedPs).then(async amountsArray => {
             const initialAllocation = arrayToObj(
               amountsArray,
               proposalKeywords,
             );
 
-            const offerResultPromiseKit = makePromiseKit();
-            // Don't trigger Node.js's UnhandledPromiseRejectionWarning.
-            // This does not suppress any error messages.
-            offerResultPromiseKit.promise.catch(_ => {});
-            const exitObjPromiseKit = makePromiseKit();
-            // Don't trigger Node.js's UnhandledPromiseRejectionWarning.
-            // This does not suppress any error messages.
-            exitObjPromiseKit.promise.catch(_ => {});
-            const seatHandle = makeHandle('SeatHandle');
-
-            const { userSeat, notifier, zoeSeatAdmin } = makeZoeSeatAdminKit(
-              initialAllocation,
-              instanceAdmin,
+            if (!invitationToZoeSeatAdmin.has(invitationHandle)) {
+              throw Error(`All paths should result in a pre-existing seat.`);
+            }
+
+            const zoeSeatAdmin = invitationToZoeSeatAdmin.get(invitationHandle);
+            const {
+              offerResultP,
+              exitObj,
+            } = await instanceAdmin.tellZCFOfferAdded(
```

Comment:
IIUC, the use you make of `exitObj` and `offerResultP` below is promise tolerant so you don't need the `await` for that reason. You can use `E.G(..)` to destructure into promises for these fields without delaying the rest of this method waiting for a round trip. 

What I don't know is if the rest of the method needs to be delayed until the `tellZCFOfferAdded` completes, in which case the `await` is fine.

```suggestion
            const {
              offerResultP,
              exitObj: exitObjP,
            } = E.G(instanceAdmin.tellZCFOfferAdded(
```